### ISSUE TITLE
Old pods not removed after rolling update

### ISSUE THREAD

**Issue by zjorgensenbits** [SeldonIO/seldon-core#1325]
I'm using the latest version of MiniKF, which includes seldon-core-operator:0.3.1.

If I deploy a simple SeldonDeployment using kubectl apply and then change, say the image name and do kubectl apply again it spins up the new pods as desire but the old pods remain active as well.  When I hit the model endpoint I can see that sometimes it's being served by one of the old pods and sometimes by the new pod. 

My deployment yaml looks like this:
```

apiVersion: machinelearning.seldon.io/v1alpha2
kind: SeldonDeployment
metadata:
  name: zachs-model
spec:
  name: zachs-test-deployment
  annotations:
    project_name: zach test
    deployment_version: v0.2
  predictors:
  - componentSpecs:
    - spec:
        containers:
        - name: classifier
          image: zjorgensenbits/seldon-core-template-model:v2
    graph:
      children: []
      endpoint:
        type: REST
      name: classifier
      type: MODEL
    name: example
    replicas: 3
    annotations:
      predictor_version: v0.3
```

**Comment by ryandawsonuk** [comment]
What status is the SeldonDeployment in if you do `kubectl get sdep zachs-model -o yaml`? The status section should be at the bottom of that output.

**Comment by zjorgensenbits** [comment]
This seems to be fixed after updating to seldon core 1.0.  The version of seldon that comes with MiniKf is apparently really old.
